otherlibs/systhreads/st_posix.h

```diff
@@ -298,6 +316,27 @@ static int st_event_wait(st_event e)
   return rc;
 }
 
+/* The tick thread: posts a SIGPREEMPTION signal periodically */
+
+static void * caml_thread_tick(void * arg)
+{
+  uintnat *domain_id = (uintnat *) arg;
+  struct timeval timeout;
+
+  caml_init_domain_self(*domain_id);
+
+  while(! Tick_thread_stop) {
```

Comment:
Shouldn't this be `atomic_load_acq` given that we are `atomic_load_store` on the variable in `caml_thread_cleanup`

Reply:
Right, good catch, just pushed a commit to address this, thank you!